src/util/plotting.jl

```diff
@@ -1,24 +1,18 @@
-@recipe f(gp::AbstractGP, x::AbstractArray) = gp(x)
-@recipe f(gp::AbstractGP, xmin::Real, xmax::Real) = gp(range(xmin, xmax; length=1_000))
-
-@recipe f(z::AbstractVector, gp::AbstractGP, x::AbstractArray) = (z, gp(x))
-@recipe function f(z::AbstractVector, gp::AbstractGP, xmin::Real, xmax::Real)
-    return (z, gp(range(xmin, xmax; length=1_000)))
-end
-
+@recipe f(x::AbstractVector, gp::AbstractGP) = gp(x)
 @recipe f(gp::FiniteGP) = (gp.x, gp)
-@recipe function f(z::AbstractVector, gp::FiniteGP)
-    length(z) == length(gp.x) ||
-        throw(DimensionMismatch("length of `z` and `gp.x` has to be equal"))
+@recipe function f(x::AbstractVector, gp::FiniteGP)
+    length(x) == length(gp.x) ||
+        throw(DimensionMismatch("length of `x` and `gp.x` has to be equal"))
+    scale = pop!(plotattributes, :ribbon_scale, 1.0)::Float64
```

Comment:
Has to be `Float64`, or would `Real` suffice?

Reply:
`Real` would be sufficient (I assume) but I wanted to avoid abstract types.

Reply:
Is there a particular reason to avoid abstract types here?

Reply:
I see. I would quite like to be able to write `ribbon_scale=3` rather than `ribbon_scale=3.0`, but I'm also not overly fussed.

Reply:
I guess we could change it to
```suggestion
    scale::Float64 = pop!(plotattributes, :ribbon_scale, 1.0)
```
instead? That would perform a conversion to `Float64` first before checking the type assertion. In fact that was what I wanted to do here but I'm always confused about where to put the type assertions :smile:

Reply:
The Int vs Float64 is a real pitfall I agree with @willtebbutt that it's probably better to have a Real.

Reply:
> That would perform a conversion to Float64 first before checking the type assertion.

I had absolutely no idea that that's how that works. I'm definitely happy for stuff to be converted to a `Float64` internally.